var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -220,6 +272,12 @@ class Cp2k(MakefilePackage, CudaPackage):
     # in the stage of cp2k@4.1
     depends_on("wannier90", when="@3.0+mpi", type="build")
 
+    with when("build_system=cmake"):
+        depends_on("dbcsr")
+        depends_on("dbcsr+openmp", when="+openmp", type=("build", "run"))
+        depends_on("dbcsr+cuda", when="+cuda", type=("build", "run"))
+        depends_on("dbcsr+rocm", when="+rocm", type=("build", "run"))
+
```

Comment:
same here, why the limitation to build and run but not link?

Reply:
my guess is that I do really understand what these options are triggering. I will correct all of these.

Reply:
Do we also have to ensure consistent settings for libxsmm?